src/main/java/org/prebid/server/bidder/ix/IxAdapter.java

```diff
@@ -64,19 +77,34 @@ private static void validatePreBidRequest(PreBidRequest preBidRequest) {
         }
     }
 
-    private BidRequest createBidRequest(AdapterRequest adapterRequest, PreBidRequestContext preBidRequestContext) {
-        final List<AdUnitBid> adUnitBids = adapterRequest.getAdUnitBids();
-
-        validateAdUnitBidsMediaTypes(adUnitBids, ALLOWED_MEDIA_TYPES);
+    private List<BidRequest> makeRequests(List<AdUnitBid> adUnitBids, PreBidRequestContext preBidRequestContext) {
+        final List<BidRequest> prioritizedRequests = new ArrayList<>();
+        final List<BidRequest> regularRequests = new ArrayList<>();
+
+        for (AdUnitBid adUnitBid : adUnitBids) {
+            final List<Format> sizes = adUnitBid.getSizes();
```

Comment:
We can optimize a little:
```
        for (AdUnitBid adUnitBid : adUnitBids) {
            boolean firstSize = true;
            for (Format size : adUnitBid.getSizes()) {
                final BidRequest bidRequest = createBidRequest(adUnitBid, size, preBidRequestContext);

                // prioritize slots over sizes
                if (firstSize) {
                    prioritizedRequests.add(bidRequest);
                    firstSize = false;
                } else {
                    regularRequests.add(bidRequest);
                }
            }
        }
```
So, rid off from calling `indexOf` on each sizes iteration.